modules/codegen/src/smithy4s/codegen/IR.scala

```diff
@@ -217,6 +237,16 @@ object TypedNode {
     case class OptionalSomeTN[A](value: A) extends FieldTN[A]
     case object OptionalNoneTN extends FieldTN[Nothing]
   }
+  sealed trait AltValueTN[+A] {
+    def map[B](f: A => B): AltValueTN[B] = this match {
+      case ProductAltTN(value) => ProductAltTN(f(value))
+      case TypeAltTN(value)    => TypeAltTN(f(value))
+    }
+  }
+  object AltValueTN {
+    case class ProductAltTN[A](value: A) extends AltValueTN[A]
+    case class TypeAltTN[A](value: A) extends AltValueTN[A]
```

Comment:
Not sure if this is the right approach, but basically I was just giving myself a way to disambiguate between the two inside of the renderer.